src/options/options.js

```diff
@@ -1,3 +1,191 @@
+    const currentInput = shortcutClearBtnElement.previousElementSibling;
+    currentInput.value = '';
+    browser.commands.update({ name: currentInput.id, shortcut: '' });
+    browser.storage.local.set({ storedShortCutsArr: [{ id: currentInput.id, shortcut: '' }] });
+    shortcutClearBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+// Reset button handler
+shortcutResetBtnArray.forEach((item) => {
+  const shortcutResetBtnElement = item;
+  shortcutResetBtnElement.addEventListener('click', () => {
+    const currentInput = shortcutResetBtnElement.parentNode.firstChild.nextSibling;
+    const getDefaultShortCutsById = defaultShortCutsArr.find((el) => el.id === currentInput.id);
+    currentInput.value = getDefaultShortCutsById.shortcut;
+    browser.commands.update({ name: currentInput.id, shortcut: currentInput.value });
+    browser.storage.local.set({ storedShortCutsArr: [{ id: currentInput.id, shortcut: currentInput.value }] });
+    shortcutResetBtnElement.parentNode.nextElementSibling.innerText = '';
+  });
+});
+
+function getOS() {
+  const { userAgent } = window.navigator;
+  const platform = window.navigator?.userAgentData?.platform || window.navigator.platform;
+  const macosPlatforms = ['Macintosh', 'MacIntel', 'MacPPC', 'Mac68K'];
+  const windowsPlatforms = ['Win32', 'Win64', 'Windows', 'WinCE'];
+  const iosPlatforms = ['iPhone', 'iPad', 'iPod'];
+
+  if (macosPlatforms.indexOf(platform) !== -1) {
+    return 'Mac OS';
+  } if (iosPlatforms.indexOf(platform) !== -1) {
+    return 'iOS';
+  } if (windowsPlatforms.indexOf(platform) !== -1) {
+    return 'Windows';
+  } if (/Android/.test(userAgent)) {
+    return 'Android';
+  } if (/Linux/.test(platform)) {
+    return 'Linux';
+  }
+
+  return null;
+}
+const isMac = getOS() === 'Mac OS';
+
+const normalizeKey = (key, keyCode) => {
+  const alphabet = /^([a-z]|[A-Z])$/;
+  if (alphabet.test(key)) return key.toUpperCase();
+
+  const digit = /^[0-9]$/;
+  const func = /^F([0-9]|1[0-2])$/;
+  const homes = /^(Home|End|PageUp|PageDown|Insert|Delete)$/;
+  if (digit.test(key) || func.test(key) || homes.test(key)) return key;
+
+  const space = /^\s$/;
+  if (space.test(key)) return 'Space';
+
+  const arrows = /^(ArrowUp|ArrowDown|ArrowLeft|ArrowRight)$/;
+  if (arrows.test(key)) return key.split('Arrow')[1];
+
+  const medias = /^(MediaPlayPause|MediaStop)$/;
+  if (medias.test(key)) return key;
+  if (key === 'MediaTrackNext') return 'MediaNextTrack';
+  if (key === 'MediaTrackPrevious') return 'MediaPrevTrack';
+
+  const keyCode0 = 48;
+  if (keyCode0 <= keyCode && keyCode <= keyCode0 + 9) return keyCode - keyCode0;
+
+  if (keyCode === 188) return 'Comma';
+  if (keyCode === 190) return 'Period';
+
+  return '';
+};
+
+function handleKeyDown(e) {
+  if (e.repeat) return;
+  if (e.key === 'Tab') {
+    window.document.activeElement.blur();
+    return;
+  }
+  const normalizedKey = normalizeKey(e.key, e.keyCode);
```

Comment:
Maybe it would be easier to use `event.code` property?

This could radically simplify implementation of the `normalizeKey()` or even lead to no need of it...

Reply:
I investigated this property. And this property does not always return the correct key value that we need to display in the input field, so it is not suitable in our case.
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code#:~:text=Be%20aware%2C%20however%2C%20that%20you%20can%27t%20use%20the%20value%20reported%20by%20KeyboardEvent.code%20to%20determine%20the%20character%20generated%20by%20the%20keystroke%2C%20because%20the%20keycode%27s%20name%20may%20not%20match%20the%20actual%20character%20that%27s%20printed%20on%20the%20key%20or%20that%27s%20generated%20by%20the%20computer%20when%20the%20key%20is%20pressed.